en/companion_computer/pixhawk_rpi.md

```diff
@@ -0,0 +1,323 @@
+
+1. Connect the Pixhawk with the laptop using a USB cable.
+1. Open QGroundControl (the vehicle should connect).
+1. [Check/change the following parameters](../advanced_config/parameters.md) in QGroundControl:
+
+   ```
+   MAV_1_CONFIG = TELEM2
+   XRCE_DDS_CFG = 0 (Disabled)
+   SER_TEL2_BAUD = 57600
+   ```
+   
+   Note that the parameters may already be set appropriately.
+   For information about how serial ports and MAVLink configuration work see [Serial Port Configuration](../peripherals/serial_configuration.md) and [MAVLink Peripherals](../peripherals/mavlink_peripherals.md).
+
+
+Then install setup MAVProxy on the RPi using the following terminal commands:
+
+1. Install MAVProxy:
+
+   ```bash
+   sudo apt install python3-pip
+   sudo pip3 install mavproxy
+   sudo apt remove modemmanager
+   ```
+
+1. Run MAVProxy, setting the port to connect to `/dev/ttyAMA0` and the baud rate to match the PX4:
+
+   ```
+   sudo mavproxy.py --master=/dev/serial0 --baudrate 57600
+   ```
+   
+   :::note
+   Note that above we used `/dev/serial0`, but we could equally well have used `/dev/ttyAMA0`.
+   If we were connecting via USB then we would instead set the port as `/dev/ttyACM0`:
+   
+   ```bash
+   sudo chmod a+rw /dev/ttyACM0
+   sudo mavproxy.py --master=/dev/ttyACM0 --baudrate 57600
+   ```
+   :::
+
+MAVProxy on RPi should now connect to the Pixhawk, via RX/TX pins.
+You should be able to see this in the RPi terminal.
+
+We have now verified that our connection is wired up properly.
+In the next section we'll set up the both Pixhawk and RPi to use XRCE-DDS and ROS2 instead of MAVLink.
+
+## ROS 2 and XRCE-DDS
+
+The [ROS 2 Guide](../ros/ros2_comm.html) and [XRCE-DDS](../middleware/xrce_dds.md) pages cover the options for setting up the XRCE-DDS and ROS, focussing on ROS 2 "Foxy".
+This tutorial uses ROS 2 "Humble" and covers the specific setup for working with RPi.
+It is worth reading both!
+
+### Pixhawk/PX4 Setup
+
+Next we set up ROS 2 instead of MAVLink on `TELEM2`.
+We do this by changing parameters in QGroundControl, which can be connected via USB, or using a telemetry radio connected to `TELEM1`.
+
+The configuration steps are:
+
+1. Connect the Pixhawk with the laptop using a USB cable and open QGroundControl (if not currently connected).
+1. [Check/change the following parameters](../advanced_config/parameters.md) in QGroundControl:
+
+   ```
+   MAV_1_CONFIG = 0 (Disabled)
+   XRCE_DDS_CFG = 102 (TELEM2)
+   SER_TEL2_BAUD = 921600
+   ```
+
+   [MAV_1_CONFIG=0](../advanced_config/parameter_reference.md#MAV_1_CONFIG) and [XRCE_DDS_CFG=102](../advanced_config/parameter_reference.md#MAV_1_CONFIG) disable MAVLink on TELEM2 and enable the XRCE-DDS client on TELEM2, respectively.
+   The `SER_TEL2_BAUD` rate sets the comms link data rate.
+   
+   Note that you could similarly configure a connection to `TELEM1`.
+1. Check that the [microdds_client](../modules/modules_system.md#microdds-client) module is now running.
```

Comment:
A reboot is always required for changes to `XRCE_DDS_CFG`, `SER_TEL2_BAUD` and `MAV_1_CONFIG` to take effect. Perhaps it should be mentioned.